Projects/Network/Sources/EndPoint/ResponseRequestable.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  ResponseRequestable.swift
+//  Network
+//
+//  Created by 리아 on 2022/10/10.
+//  Copyright © 2022 zesty. All rights reserved.
+//
+
+import Foundation
+
+protocol ResponseRequestable {
+    var scheme: String { get }
+    var host: String { get }
+    var path: String { get }
+    var method: HttpMethod { get }
+    var queryParams: Encodable? { get }
+    var bodyParams: Encodable? { get }
+    var headers: [String: String]? { get }
+}
+
+extension ResponseRequestable {
+
+    func getUrlRequest() throws -> URLRequest {
+        let url = try url()
+        var urlRequest = URLRequest(url: url)
+
+        // httpBody
+        if let bodyParams = try bodyParams?.toDictionary() {
+            if !bodyParams.isEmpty {
+                urlRequest.httpBody = try? JSONSerialization.data(withJSONObject: bodyParams)
+            }
+        }
+        // httpMethod
+        urlRequest.httpMethod = method.rawValue
+        // header
+        headers?.forEach {
+            urlRequest.setValue($1, forHTTPHeaderField: $0)
+        }
+
+        return urlRequest
+    }
+
+    private func url() throws -> URL {
+        var components = URLComponents()
+
+        components.scheme = scheme
+        components.host = host
+        components.path = path
+        components.queryItems = try getQueryItems()
+
+        guard let url = components.url else { throw NetworkError.urlComponent }
+        return url
+    }
+
+    private func getQueryItems() throws -> [URLQueryItem]? {
+        if let queryParams = try queryParams?.toDictionary() {
+            if !queryParams.isEmpty {
+                return queryParams.map { URLQueryItem(name: $0.key, value: "\($0.value)") }
+            }
+        }
+        return nil
+    }
+
+}
+
+extension Encodable {
+
+    func toDictionary() throws -> [String: Any]? {
+        let data = try JSONEncoder().encode(self)
+        let jsonData = try JSONSerialization.jsonObject(with: data)
+        return jsonData as? [String: Any]
```

Comment:
p3
옵셔널 처리를 해보는건 어떨까요??

```suggestion
guard let data = try? JSONEncoder().encode(self) else { return [:] }
guard let dictionary = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any] else { return [:] }
return dictionary
```

Reply:
현재 try 코드들은 모두 throw를 통해서 에러가 `NetworkServable`까지 전달되기 때문에, 
`RequestReponsable` 의 모든 코드는 에러를 뿜게 작성했습니다.

제안해주신 대로 옵셔널 바인딩을 하면 `return [:]` 같이 의미없는 빈 값을 전달하게 되는데,
네트워크 통신 과정 상에서는 빈 값을 전달하기 보다는 확실하게 실패했다는 에러를 전달하는 것이 더 중요하다고 저는 생각합니다!